tldw_Server_API/app/api/v1/endpoints/admin/__init__.py

```
@@ -0,0 +1,2014 @@
+
+@router.post("/roles/{role_id}/permissions/tools", response_model=ToolPermissionResponse)
+async def grant_tool_permission_to_role(role_id: int, payload: ToolPermissionGrantRequest, db=Depends(get_db_transaction)) -> ToolPermissionResponse:
+    """Grant a tool execution permission to a role.
+
+    - tool_name='*' → grants tools.execute:*
+    - tool_name='<name>' → grants tools.execute:<name>
+    Creates the permission in catalog if missing.
+    """
+    tool = payload.tool_name.strip()
+    name = f"tools.execute:{'*' if tool == '*' else tool}"
+    desc = f"Wildcard tool execution" if tool == '*' else f"Execute tool {tool}"
```

Ruff (0.14.14)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)